libtock/tock.c

```diff
@@ -520,48 +528,87 @@ void* memop(uint32_t op_type, int arg1) {
     : "r" (a0), "r" (a1)
     : "memory"
     );
-  if (code == TOCK_SYSCALL_SUCCESS_U32) {
-    return val;
+  if (code == TOCK_SYSCALL_SUCCESS) {
+    memop_return_t rv = {TOCK_STATUSCODE_SUCCESS, 0};
+    return rv;
+  } else if (code == TOCK_SYSCALL_SUCCESS_U32) {
+    memop_return_t rv = {TOCK_STATUSCODE_SUCCESS, val};
+    return rv;
+  } else if (code == TOCK_SYSCALL_FAILURE) {
+    memop_return_t rv = {(statuscode_t) val, 0};
+    return rv;
   } else {
-    return NULL;
+    // Invalid return type
+    exit(1);
```

Comment:
Should we return a failure instead of `exit()`?

Reply:
The other syscalls exit. If this happens the kernel is broken and there isn't really anything a process can do.